appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/annotation/impl/AppClientScanner.java

```diff
@@ -158,4 +167,34 @@ private URI scanURI(URI uriToAdd) throws IOException {
         }
         return uriToAdd;
     }
+
+    /**
+     * Overriding to handle the case where EJB class is mistakenly packages inside an appclient jar.
+     * Instead of throwing an error which might raise backward compatiability issues, a cleaner way is to
+     * just skip the annotation processing for them.
+     */
+    @Override
+    protected void calculateResults(ApplicationClientDescriptor bundleDesc) {
+        super.calculateResults(bundleDesc);
+
+        Class[] ejbAnnotations = ejbProvider.getAnnotationTypes();
```

Comment:
The impl in AbstractHandler just constructs an empty array with zero element if ejbProvider is null. In our case, if it is null, we could just return (as the changes worked for that test case, I am assuming in the normal cases, it will not be null. The null check just for some unexpected code paths). Alternatively, we can go to back to the hard-coded strings if ejbProvider is null.